sa/database.go

```diff
@@ -12,11 +13,33 @@ import (
 	blog "github.com/letsencrypt/boulder/log"
 )
 
+// DbSettings contains tuning settings for the database/sql driver. The
+// zero value of each field means use the default setting from
+// database/sql. ConnMaxIdleTime and ConnMaxLifetime should be set lower
+// than their mariab counterparts interactive_timeout and wait_timeout.
+type DbSettings struct {
+	// MaxOpenConns sets the maximum number of open connections to the
+	// database. If left unspecified connections to the database default
+	// to unlimited.
+	MaxOpenConns int
+	// MaxIdleConns is optional. When n > 0 it sets the size of the idle
+	// connection pool. The value is a pointer to an int because if n <=
+	// 0, no idle connections are retained. If nil is specified the sql
+	// driver will use the package default instead.
+	MaxIdleConns *int
```

Comment:
Using `*int` here makes sense for the reason you explain, but I'd like to push back a little bit.

The way I see it, there are two main questions informing how we use this config parameter:
1) How often do we want to set it to a non-default value at all?
2) How often do we want to set it specifically to "no idle connections retained"?

Let's examine this 2x2 grid of possibilities:
rare/rare: in this case, it makes sense to just use an int: most of the time the config can have a default value of 0, which our code here can interpret as "use the package default", most of our few overrides will be positive numbers that make sense, and for our tiny number of overrides where we want to turn off idle connections all together we can put a negative number in the config.
common/rare: same as above: we'll have a bunch of configs with various positive values, but only a few that want to retain no idle connections, so those ones can use negative numbers.
rare/common: given that we're only rarely overriding the default at all, setting a bunch of those overrides to be -1 to mean "retain no idle connections" doesn't seem like a hardship
common/common: this is the situation in which using an `*int` makes sense, so we're not just littering every single config with negative numbers.

I don't think that we're in the "common/common" case, so I think it's fine to use just a plain int, assume that a value of 0 means "use the package default", and require configs to have a negative number in order to mean "don't retain any idle connections".